### PR TITLE
[WIP] Fix for #351 feature OSD-SDK

### PR DESCRIPTION
**Pull request by ram-infrac**
Signed-off-by: Ram Suradkar ram.suradkar@portworx.com

**What this PR does / why we need it**:
This PR is for asking suggestion for implementation approach #351. Please do not merge 
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/server/cluster.go [hunk 1]
```diff
@@ -40,6 +41,7 @@ func (c *clusterApi) Routes() []*Route {
 		{verb: "PUT", path: clusterPath("/shutdown/{id}", cluster.APIVersion), fn: c.shutdown},
 		{verb: "GET", path: clusterPath("/alerts/{resource}", cluster.APIVersion), fn: c.enumerateAlerts},
 		{verb: "PUT", path: clusterPath("/alerts/{resource}/{id}", cluster.APIVersion), fn: c.clearAlert},
+		{verb: "POST", path: clusterPath("/setclustersecretkey", cluster.APIVersion), fn: c.setClusterSecretKey},
```
Comment (ram-infrac) [inline api/server/cluster.go]: Should this go into api/server/secrets.go , where secret.go will implement restserver interface from api/server/server.go

File: cluster/cluster.go [hunk 2]
```diff
@@ -267,6 +267,16 @@ type Cluster interface {
 	ClusterAlerts
 }
 
+//Secrets generic secret store operations
```
Comment (ram-infrac) [inline cluster/cluster.go]: Should I create separate package structure for each of PX API (secrets, schedule policy, service cmds etc) section moving into OSD or should they be clubbed together.

File: cluster/cluster.go [hunk 3]
```diff
@@ -267,6 +267,16 @@ type Cluster interface {
...
+type Secrets interface {
```
Comment (ram-infrac) [inline cluster/cluster.go]: should I create separate package for each of PX API sections(service, diags command) which will contain interface and there implementation, or should all those clubbed together

Comment (lpabon) [reply cluster/cluster.go]: I think it is ok to have the interface here.

File: api/server/cluster.go [hunk 4]
```diff
@@ -648,6 +650,23 @@ func (c *clusterApi) eraseAlert(w http.ResponseWriter, r *http.Request) {
...
+	method := "setClustersecretKey"
+	if err := json.NewDecoder(r.Body).Decode(&dcReq); err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	//Logged recived paramss for now
+	fmt.Println("clusterkey", dcReq.Clustersecretkey, "override", dcReq.Override)
+
+	w.Write([]byte(secretKeyOkMsg + "\n"))
```
Comment (lpabon) [inline api/server/cluster.go]: Seems like these lines are just for debugging, right?

Comment (ram-infrac) [reply api/server/cluster.go]: yes Luis those are for debugging purpose, I carved out code for /setclustersecretkey endpoint and put this PR request to clarify approach.

**Pull request closed without merging**